compiler/src/dotty/tools/dotc/transform/localopt/StringInterpolatorOpt.scala

```diff
@@ -36,7 +36,7 @@ class StringInterpolatorOpt extends MiniPhase {
     def unapply(tree: Select)(implicit ctx: Context): Boolean = {
       tree.symbol.eq(defn.StringContextModule_apply) && {
         val qualifier = tree.qualifier
-        qualifier.isInstanceOf[Ident] && qualifier.symbol.eq(defn.StringContextModule)
+        qualifier.isInstanceOf[Select] && qualifier.symbol.eq(defn.StringContextModule)
```

Comment:
Thinking about it, the first test might not be needed. The method could be rewritten as:
```scala
def unapply(tree: Select)(implicit ctx: Context): Boolean =
  tree.symbol.eq(defn.StringContextModule_apply) &&
  tree.qualifier.symbol.eq(defn.StringContextModule)
```

Reply:
Indeed, it seemed a bit fragile. Removed.